test/expr/unary/keypath/keypath.swift

```diff
@@ -230,6 +230,63 @@ func testKeyPathSubscript(readonly: Z, writable: inout Z,
   writable[keyPath: akp] = anyqSink2 // expected-error{{cannot assign to immutable}}
 }
 
+struct ZwithSubscript {
+  subscript(keyPath: KeyPath<ZwithSubscript, Int>) -> Int { return 0 }
+  subscript(keyPath: WritableKeyPath<ZwithSubscript, Int>) -> Int { return 0 }
+  subscript(keyPath: ReferenceWritableKeyPath<ZwithSubscript, Int>) -> Int { return 0 }
+  subscript(keyPath: PartialKeyPath<ZwithSubscript>) -> Any { return 0 }
```

Comment:
Subscripts don't have significant labeled arguments unless you explicitly ask for them, so all of these declarations really declare `subscript(_:)`. You'd have to declare `subscript(keyPath kp: ...)` to test overloading.